### DISCUSSION TITLE
What --extensions do for babel-node

### DISCUSSION THREAD

**Discussion by mukuljainx** [babel/babel#14388] [General] [answered]
There was an `import` statement in my `ts` file, without this argument it was giving syntax error but after 
`babel-node --extensions '.ts'` it worked, what does it do?

I tried to find about in doc, but 

>List of extensions to hook into

is not clear enough.

Any help would be great, thankyou!

**Comment by JLHwung** [comment, marked as answer]
> List of extensions to hook into

When users are calling `require("some-module.ts")`, the Node.js cjs loader will call the hook registered by `@babel/node` according to the `extensions` config, so that Babel gets a chance to transpile user codes before they are executed. See also https://nodejs.org/api/modules.html#requireextensions

**Reply by mukuljainx** [reply]
Thank you @JLHwung for the insight!
